### PR TITLE
fix: comparison view parallel coordinates chart shouldn't break when selecting rows [ET-261]

### PR DESCRIPTION
**Pull request by EmilyBonar**
## Ticket

[ET-261]

## Test Plan

Go to the Glide experiment list, select an experiment, open comparison view, and switch to the Hyperparameters tab to view the Parallel Coordinates plot. From there, select and unselect various experiments with different hyperparameters and metrics. At no point should the selected metric change unless you unselect the only experiment that has it. At no point should all of the lines on the chart disappear.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[ET-261]: https://hpe-aiatscale.atlassian.net/browse/ET-261?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by johnkim-det** [comment]
testing on netlify, I think I'm still seeing some cases where selecting additional experiments or runs removes previously charted ones. are there cases where we would still expect something like this to happen?


https://github.com/determined-ai/determined/assets/97752292/f3b8a647-0dfc-43e3-8010-403ac4501f88

https://github.com/determined-ai/determined/assets/97752292/59a4f070-07a1-420a-84c4-a9e7e6a6e836

**Review by johnkim-det — COMMENTED** [review]
Looks like there are still cases where this happens

https://github.com/determined-ai/determined/assets/97752292/f1c39f0c-0181-4dd7-89ce-06e800a823a6

https://github.com/determined-ai/determined/assets/97752292/d0ca9554-2efc-407d-aa80-bc311a34d9dd

https://github.com/determined-ai/determined/assets/97752292/7e514de9-3cef-4a7a-9ee0-782ced5daea1

https://github.com/determined-ai/determined/assets/97752292/707640dd-d1b9-474e-8416-10e3e7e92ecc

Also not sure if this state is expected:

Screenshot 2024-07-02 at 2 45 47 PM

**Comment by EmilyBonar** [comment]
@johnkim-det Ok try it again. I'll remove the console.logs once we can confirm it works. And yes, that message is expected in the last screenshot.

File: webui/react/src/components/CompareParallelCoordinates.tsx [hunk 1]
```diff
@@ -90,17 +89,17 @@ const CompareParallelCoordinates: React.FC<Props> = ({
...
         },
         dimension: { label: { angle: Math.PI / 4, truncate: 24 } },
         padding: [4, 120, 4, 16],
       },
     }),
-    [colorMap, hermesCreatedFilters, selectedMetric, chartData?.trialIds],
+    [colorMap, hermesCreatedFilters, selectedMetric, chartData?.recordIds],
   );
 
   const dimensions = useMemo(() => {
```
Comment (johnkim-det) [inline webui/react/src/components/CompareParallelCoordinates.tsx:105]: question: not necessarily related to your changes, but is `dimensions` working as intended with Runs + Categorical/Log HP types? I think this is dependent on `experimentHyperparameters` having `selectedExperiments`, but I could be missing something.

File: webui/react/src/components/CompareParallelCoordinates.tsx [hunk 2]
```diff
@@ -148,80 +147,86 @@ const CompareParallelCoordinates: React.FC<Props> = ({
...
   ]);
 
+  const extractHyperparams = useCallback(
+    (
+      record: TrialItem | FlatRun,
+      metricsMap: Record<number, number>,
+      hpMap: Record<string, Record<number, Primitive>>,
+    ) => {
+      if (!selectedMetric) return;
+      const idKey = isRun(record) ? record.id : record.experimentId;
```
Comment (johnkim-det) [inline webui/react/src/components/CompareParallelCoordinates.tsx:157]: nit: might make it slightly easier to remove experiments-specific code in the future to comment the parts that are dependent on `selectedExperiments`/`trials` vs `selectedRuns` props.

**Review by johnkim-det — APPROVED** [review]
Having trouble testing with Runs, but Experiments looks good! A couple comments but as long as Runs looks good in your testing I think this is good to merge.

File: webui/react/src/components/CompareParallelCoordinates.tsx (hunk 1 above)
Comment (EmilyBonar) [reply webui/react/src/components/CompareParallelCoordinates.tsx:105]: That's a good idea for future work! Right now runs api return hyperparameters as `Record<string, HpValue>`, so there's no hp type to do anything with.

**Pull request merged**